jdi-light/src/main/java/com/epam/jdi/light/elements/base/JDIBase2.java

```diff
@@ -0,0 +1,525 @@
+        if (locator == null) return null;
+        return correctXPaths(locator);
+    }
+
+    public String printContext() {
+        JDIBase jdiBase = getBase(parent);
+        if (jdiBase == null) return "";
+        String locator = jdiBase.getLocator() == null
+                ? ""
+                : jdiBase.locator.printLocator();
+        if (jdiBase.parent == null)
+            return locator;
+        if (isBlank(locator))
+            return jdiBase.printContext();
+        return jdiBase.printContext() + ">" + locator;
+    }
+    public String printFullLocator() {
+        return parent == null || isBlank(printContext())
+                ? locator.printLocator()
+                : printContext() + ">" + locator.printLocator();
+    }
+    private void initContext() {
+        context = printFullLocator();
+    }
+
+    @Override
+    public String toString() {
+        initContext();
+        try {
+            return PRINT_ELEMENT.execute(this);
+        } catch (Exception ex) { throw exception(ex, "Can't print element"); }
+    }
+    private static String printWebElement(WebElement element) {
+        String asString = element.toString().replaceAll("css selector", "css");
+        String result = asString.startsWith("WebElement->")
+                ? "" : "WebElement->";
+        if (asString.contains(")]")) {
+            String s = asString.split("-> ")[1];
+            return result + s.substring(0,s.length()-1);
+        }
+        return asString;
+    }
+
+    public static JFunc1<JDIBase, String> PRINT_ELEMENT = element -> {
+        if (element.webElement.hasValue())
+            return printWebElement(element.webElement.get());
+        return Switch(LOGS.logLevel).get(
+                Case(l -> l == STEP,
+                    l -> msgFormat(PRINT_ELEMENT_STEP, element)),
+                Case(l -> l == INFO,
+                    l -> msgFormat(PRINT_ELEMENT_INFO, element)),
+                Case(l -> l == ERROR,
+                    l -> msgFormat(PRINT_ERROR_STEP, element)),
+                Default(l -> msgFormat(PRINT_ELEMENT_DEBUG, element))
+        );
+    };
+
+    public void actions(JFunc2<Actions, WebElement, Actions> action) {
+        action.execute(actions.get(), get()).build().perform();
+    }
+    public void actions(JFunc1<Actions, Actions> action) {
+        action.execute(actions.get()).build().perform();
+    }
+    public void actionsWithElement(JFunc2<Actions, WebElement, Actions> action) {
+        action.execute(actions.get().moveToElement(get()), get()).build().perform();
+    }
+    public void actionsWithElement(JFunc1<Actions, Actions> action) {
+        action.execute(actions.get().moveToElement(get())).build().perform();
+    }
+    private Safe<Actions> actions = new Safe<>(() -> new Actions(driver()));
+
+    public ElementArea clickAreaType = ELEMENT.clickType;
+    public TextTypes textType = ELEMENT.getTextType;
+    public SetTextTypes setTextType = ELEMENT.setTextType;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5327794)